### PR TITLE
Responsive radio

### PR DESCRIPTION
**Pull request by juanfabrega**
# Github Issue or Trello Card
This PR addresses this issue: https://github.com/palmetto/palmetto-components/issues/437

# What type of change is this?
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updating documentation.
- [ ] Updating deployment/build pipeline.
- [ ] Updating Project Dependencies
- [ ] Improving or adding to Test Coverage

# Completeness Checklist

- [X] TESTS: My changes maintain the baseline required test coverage, as specified by code climate analysis.
- [X] DOCS: All new component work is covered in that component's `Overview` docs.
- [X] DOCS: All new component work is covered in a component `Playground`.
- [X] DOCS: All new visual changes or options are covered under relevant components' `VisualTests`.
- [X] My code has no linting or typescript compile warnings.
- [X] My work is tied to a Github issue and satisfies the acceptance criteria (if applicable) of the corresponding issue.

# Quality Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

# UI Checklist
- [X] I have conducted visual UAT on my changes/features.
- [X] My solution works well on desktop, tablet, and mobile browsers.

### CODE REVIEW

File: src/components/RadioGroup/RadioGroup.Overview.stories.mdx [hunk 1]
```diff
@@ -320,6 +321,67 @@ Use the `error` prop to mark the input as invalid. `error` accepts a `boolean`,
...
+            onChange={event => handleChange(event, 'lg')}
+            value={state.lg}
+            options={options}
+            size="lg"
+          />
+          <RadioGroup
+            title="Responsive"
+            name="responsiveRadio"
+            onChange={event => handleChange(event, 'responsive')}
+            value={state.lg}
```
Comment (nathanyoung) [inline src/components/RadioGroup/RadioGroup.Overview.stories.mdx:375]:
```suggestion
            value={state.responsive}
```

**Review by nathanyoung — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/1447339/135641631-8c2ad99d-a97c-4fc9-949e-d087ca47130e.mov

I think this is caused because the input ids aren't unique.

**Comment by juanfabrega** [comment]
@nathanyoung can you still replicate the issue with mixed state across inputs? I am unable to replicate it at the moment.

**Comment by nathanyoung** [comment]
> @nathanyoung can you still replicate the issue with mixed state across inputs? I am unable to replicate it at the moment.

yes, it's because the ids are not unique. There are accessibility warnings too. 

Screen Shot 2021-10-04 at 8 27 21 AM

**Comment by nathanyoung** [comment]
[Responsive sizes design and implementation for RadioGroup](https://trello.com/c/TYgQyo8r/737-responsive-sizes-design-and-implementation-for-radiogroup)

**Review by nathanyoung — APPROVED** [review]

**Pull request merged**